### PR TITLE
[CUTEst] rebuild with CSL

### PR DESCRIPTION
**Pull request by dpo**

### CODE REVIEW

File: C/CUTEst/build_tarballs.jl [hunk 1]
```diff
@@ -3,7 +3,7 @@
 using BinaryBuilder
 
 name = "CUTEst"
-version = v"0.2.0"
+version = v"0.2.3"
```
Comment (giordano) [inline C/CUTEst/build_tarballs.jl:6]: Why is the version changing if the source is the same?

Comment (dpo) [reply C/CUTEst/build_tarballs.jl:6]: I was confused the first time I submitted this and thought `version` was the version of the recipe. Shall I leave it as it was?

Comment (giordano) [reply C/CUTEst/build_tarballs.jl:6]: Well, ideally the version should be that of the upstream package

Comment (dpo) [reply C/CUTEst/build_tarballs.jl:6]: Duh. I wanted to write 2.0.3. Let me fix it.

File: C/CUTEst/build_tarballs.jl [hunk 2]
```diff
@@ -3,16 +3,16 @@
...
-    "https://github.com/ralna/SIFDecode/archive/v2.0.3.tar.gz" =>
-    "3a4aa817e1bf4e3595d0e4378da6172b65f02861f3a7c39f9da632a5cc31b1b2",
-    "https://github.com/ralna/CUTEst/archive/v2.0.3.tar.gz" =>
-    "d21a65c975302296f9856c09034cf46edc5da34b6efd96eed6cc94af6d2c8a55",
+    ArchiveSource("https://github.com/ralna/ARCHDefs/archive/v2.0.3x.tar.gz",
+                  "6583e27f84338447767bbdf4335514c8836ae4ad54f5e66280307e8b57189cff"),
+    ArchiveSource("https://github.com/ralna/SIFDecode/archive/v2.0.3.tar.gz",
+                  "3a4aa817e1bf4e3595d0e4378da6172b65f02861f3a7c39f9da632a5cc31b1b2"),
+    ArchiveSource("https://github.com/ralna/CUTEst/archive/v2.0.3.tar.gz",
+                  "d21a65c975302296f9856c09034cf46edc5da34b6efd96eed6cc94af6d2c8a55"),
```
Comment (giordano) [inline C/CUTEst/build_tarballs.jl:15]: I used the new `ArchiveSource` type to fix the warnings that were flooding the summary of the build

File: C/CUTEst/build_tarballs.jl [hunk 3]
```diff
@@ -124,6 +109,7 @@ products = [
 ]
 
 dependencies = [
+    Dependency("CompilerSupportLibraries_jll"),
```
Comment (giordano) [inline C/CUTEst/build_tarballs.jl:112]: CSL is also needed at runtime, it's actually one of the main points of existence of this package.

File: C/CUTEst/build_tarballs.jl (hunk 2 above)
Comment (dpo) [reply C/CUTEst/build_tarballs.jl:15]: Thank you. Sorry I should be more careful.

**Pull request merged**